MapboxVision/src/main/java/com/mapbox/vision/session/SessionManager.kt

```diff
@@ -0,0 +1,255 @@
+import com.mapbox.vision.mobile.core.utils.extentions.TAG_CLASS
+import com.mapbox.vision.mobile.core.utils.extentions.lazyUnsafe
+import com.mapbox.vision.models.video.VideoCombined
+import com.mapbox.vision.sync.SyncClient
+import com.mapbox.vision.sync.filemanager.SyncDirectoriesProvider
+import com.mapbox.vision.sync.filemanager.SyncFileHandler
+import com.mapbox.vision.sync.telemetry.TelemetryImageSaverImpl
+import com.mapbox.vision.sync.telemetry.TelemetryMetaGenerator
+import com.mapbox.vision.sync.telemetry.TelemetryQueue
+import com.mapbox.vision.sync.telemetry.TelemetrySyncManager
+import com.mapbox.vision.sync.util.TelemetryEnvironment
+import com.mapbox.vision.sync.util.VideoMetadataJsonMapper
+import com.mapbox.vision.sync.util.VisionProEnvironment
+import com.mapbox.vision.sync.visionpro.VisionProMetaGenerator
+import com.mapbox.vision.sync.visionpro.VisionProQueue
+import com.mapbox.vision.sync.visionpro.VisionProSyncManager
+import com.mapbox.vision.utils.FileUtils
+import com.mapbox.vision.utils.VisionLogger
+import com.mapbox.vision.utils.file.RotatedBuffers
+import com.mapbox.vision.video.videoprocessor.VideoProcessor
+import com.mapbox.vision.video.videosource.camera.VideoRecorder
+import okhttp3.OkHttpClient
+
+internal interface SessionManager {
+
+    fun start()
+
+    fun stop()
+
+    fun setCountry(country: Country)
+
+    fun startRecording(path: String)
+
+    fun stopRecording()
+
+    class Impl(
+        private val application: Application,
+        private val nativeVisionManager: NativeVisionManager,
+        private val okHttpClient: OkHttpClient,
+        private val videoRecorder: VideoRecorder,
+        mapboxTelemetry: MapboxTelemetry,
+        private val telemetryImageSaver: TelemetryImageSaver,
+        gson: Gson
+    ) : SessionManager {
+
+        companion object {
+            private const val CACHE_DIR = "Cache"
+            private const val VIDEO_BUFFERS_DIR = "Buffers"
+        }
+
+        private lateinit var sessionWriter: SessionWriter
+        private var currentCountry = Country.Unknown
+        private var isRecording = false
+        private val videoProcessor: VideoProcessor
+
+        private val syncManagers by lazyUnsafe {
+            arrayOf(telemetrySyncManager, visionProSyncManager)
+        }
+
+        private val countryChangeListeners = ArrayList<EnvironmentSettings.CountryChange>()
+
+        private val telemetrySyncManager: TelemetrySyncManager
+        private val visionProSyncManager: VisionProSyncManager
+
+        private val telemetrySession: EnvironmentSession<EnvironmentSession.Telemetry.EnvironmentData>
+        private val visionProSession: EnvironmentSession<EnvironmentSession.VisionPro.EnvironmentData>
+
+        init {
+            videoProcessor = VideoProcessor.Impl()
+
+            val videoMetadataJsonMapper = VideoMetadataJsonMapper.Impl(gson)
+
+            val telemetrySyncFileProvider = SyncDirectoriesProvider.Impl(
```

Comment:
maybe add for each *Session function for initialization?

Reply:
didn't get your point